### PR TITLE
add travis test

### PR DESCRIPTION
**Pull request by Linary**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+jdk:
+  - oraclejdk8
+
+sudo: required
+
+install: mvn compile -Dmaven.javadoc.skip=true
+
+before_script:
+  - echo -e "Host heroku.com\n\tStrictHostKeyChecking no\n" >> ~/.ssh/config
+  - echo -e "Host github.com\n\tStrictHostKeyChecking no\n" >> ~/.ssh/config
```
Comment (javeme) [inline .travis.yml]: are the two line same?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  - $TRAVIS_DIR/install-hugegraph.sh
+
+script: mvn test -Dtest=LoaderTest && mvn cobertura:cobertura
+
+after_success:
+  - bash <(curl -s https://codecov.io/bash)
+
+env:
+  global:
+  - TRAVIS_DIR=assembly/travis
```
Comment (javeme) [inline .travis.yml]: add "\n"

File: README.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # hugegraph-loader
 
+[![License](https://img.shields.io/badge/license-Apache%202-0E78BA.svg)](https://www.apache.org/licenses/LICENSE-2.0.html)
+[![Build Status](https://travis-ci.org/hugegraph/hugegraph-loader.svg?branch=master)](https://travis-ci.org/hugegraph/hugegraph-loader)
```
Comment (javeme) [inline README.md]: status -- keep consistent with core

File: README.md [hunk 4]
```diff
@@ -1,5 +1,9 @@
 # hugegraph-loader
 
+[![License](https://img.shields.io/badge/license-Apache%202-0E78BA.svg)](https://www.apache.org/licenses/LICENSE-2.0.html)
```
Comment (javeme) [inline README.md]: licence

File: example/struct.json [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        "path": "example/edge_created.json",
+        "format": "JSON"
+      },
+      "mapping": {
+        "aname": "name",
+        "bname": "name"
+      }
+    }
+  ]
+}
```
Comment (javeme) [inline example/struct.json]: ditto

File: README.md (hunk 3 above)
Comment (Linary) [reply README.md]: core is just like this.

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+jdk:
+  - oraclejdk8
+
+sudo: required
+
+install: mvn compile -Dmaven.javadoc.skip=true
+
+before_script:
+  - $TRAVIS_DIR/install-hugegraph.sh
```
Comment (javeme) [inline .travis.yml]: Install the corresponding version instead of the latest one

Comment (javeme) [inline .travis.yml]: install the corresponding version instead of the latest one

Comment (javeme) [inline .travis.yml]: Install corresponding version instead of the latest one

Comment (javeme) [inline .travis.yml]: Install corresponding version instead of the latest one

Comment (javeme) [inline .travis.yml]: Install corresponding version instead of the latest one

Comment (javeme) [inline .travis.yml]: Install corresponding version instead of the latest one

Comment (javeme) [inline .travis.yml]: Install corresponding version instead of the latest one

Comment (javeme) [inline .travis.yml]: Install corresponding version instead of the latest one

Comment (javeme) [inline .travis.yml]: Install corresponding version instead of the latest one

Comment (javeme) [inline .travis.yml]: Install corresponding version instead of the latest one

Comment (javeme) [inline .travis.yml]: Install corresponding version instead of the latest one

Comment (javeme) [inline .travis.yml]: Install corresponding version instead of the latest one

**Comment by javeme** [comment]
Install corresponding version instead of the latest one

**Comment by javeme** [comment]
Install corresponding version instead of the latest one

**Comment by javeme** [comment]
Install corresponding version instead of the latest one

**Comment by javeme** [comment]
Install corresponding version instead of the latest one

**Comment by javeme** [comment]
Install corresponding version instead of the latest one

**Comment by javeme** [comment]
Github Incident: https://blog.github.com/2018-10-21-october21-incident-report/

File: example/edge_created.json [hunk 7]
```diff
@@ -0,0 +1,4 @@
+{"aname": "marko", "bname": "lop", "date": "20171210", "weight": 0.4}
```
Comment (javeme) [inline example/edge_created.json]: prefer to change the test dir

File: assembly/travis/install-hugegraph.sh [hunk 8]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+set -ev
+
+LOADER_VERSION=`cat "pom.xml" | grep "<version>" | head -1 | awk -F '<|>' '{print $3}'`
```
Comment (javeme) [inline assembly/travis/install-hugegraph.sh]: pass current branch here to determine if using the latest version or release version

File: src/test/java/com/baidu/hugegraph/loader/test/functional/FileUtil.java [hunk 9]
```diff
@@ -50,6 +51,12 @@ public static void append(String fileName, String line) {
         append(fileName, line, DEFAULT_CHARSET);
     }
 
+    public static void append(String fileName, List<String> lines) {
```
Comment (javeme) [inline src/test/java/com/baidu/hugegraph/loader/test/functional/FileUtil.java]: change to append(String fileName, String... lines)
and call:
```
 FileUtil.append(path("vertex_person.csv"), 
                 "line-1",
                 "line-2")
```

File: src/test/java/com/baidu/hugegraph/loader/test/functional/FileUtil.java [hunk 10]
```diff
@@ -46,11 +46,14 @@ public static void clear(String fileName) {
...
-        append(fileName, line, DEFAULT_CHARSET);
+    public static void append(String fileName, String... lines) {
+        for (String line : lines) {
+            appendInCharset(fileName, line, DEFAULT_CHARSET);
+        }
     }
 
-    public static void append(String fileName, String line, String charset) {
+    public static void appendInCharset(String fileName, String line,
+                                       String charset) {
```
Comment (javeme) [inline src/test/java/com/baidu/hugegraph/loader/test/functional/FileUtil.java]: append(String fileName, Charset charset, String... lines)

use Arrays.asList(lines) for writeLines
get Charset from Charsets like Charsets.UTF_8

File: src/test/java/com/baidu/hugegraph/loader/test/functional/FileUtil.java [hunk 11]
```diff
@@ -46,11 +46,14 @@ public static void clear(String fileName) {
         }
     }
 
-    public static void append(String fileName, String line) {
-        append(fileName, line, DEFAULT_CHARSET);
+    public static void append(String fileName, String... lines) {
```
Comment (javeme) [inline src/test/java/com/baidu/hugegraph/loader/test/functional/FileUtil.java]: append(String fileName, String... lines)

File: src/test/java/com/baidu/hugegraph/loader/test/functional/FileUtil.java [hunk 12]
```diff
@@ -46,19 +47,21 @@ public static void clear(String fileName) {
...
```
Comment (javeme) [inline src/test/java/com/baidu/hugegraph/loader/test/functional/FileUtil.java]: should update reference like:
FileUtil.append(path("vertex_software.csv"), line, "GBK")

File: src/test/java/com/baidu/hugegraph/loader/test/functional/FileUtil.java [hunk 13]
```diff
@@ -46,19 +47,21 @@ public static void clear(String fileName) {
...
     }
 
-    public static void append(String fileName, String line, String charset) {
+    public static void append(String fileName, Charset charset,
+                              String... lines) {
         File file = org.apache.commons.io.FileUtils.getFile(fileName);
         checkFileValid(file, true);
         try {
-            FileUtils.writeLines(file, charset, ImmutableList.of(line), true);
+            FileUtils.writeLines(file, charset.name(), Arrays.asList(lines),
```
Comment (javeme) [inline src/test/java/com/baidu/hugegraph/loader/test/functional/FileUtil.java]:
```
FileUtils.writeLines(file, charset.name(),
                     Arrays.asList(lines), true)
```

**Pull request merged**